.github/workflows/deploy.yml

```
@@ -0,0 +1,76 @@
+name: Deploy contracts
+
+on:
+  workflow_dispatch: # Manual trigger
+    inputs:
+      network:
+        description: 'Network'
+        required: true
+        type: choice
+        options:
+          - hardhat
+          - bellecour
+          - arbitrumSepolia
+          - arbitrum
+        default: 'hardhat'
+      whitelist_owner:
+        description: 'Whitelist contract owner address (optional - defaults to deployer)'
+        required: false
+        type: string
+
+jobs:
+
+  # Build and test before deploying.
+  build-and-test:
+    uses: ./.github/workflows/main.yml
+
+  # Deploy and verify contract.
+  deploy:
+    needs: build-and-test
+    runs-on: ubuntu-latest
+    env:
+      CI: true
+    permissions:
+      contents: write # Required to commit deployment files.
+    environment: ${{ inputs.network }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Deploy contracts
+        env:
+          WHITELIST_OWNER: ${{ inputs.whitelist_owner }}
+          # For Deployment
+          HARDHAT_IGNITION_CONFIRM_DEPLOYMENT: false
+          PRIVATE_KEY: ${{ secrets.DEPLOYER_PRIVATE_KEY }}
+          RPC_URL: ${{ secrets.RPC_URL }}
+        run: |
+            npm run deploy -- --network ${{ inputs.network }} --deployment-id ${{ inputs.network }}
+
+      - name: Verify contracts
+        if: inputs.network != 'hardhat'
+        continue-on-error: true
+        env:
+          # For Verification
+          EXPLORER_API_KEY: ${{ secrets.EXPLORER_API_KEY }}
+          IS_VERIFICATION_API_V2: ${{ vars.IS_VERIFICATION_API_V2  }}
+        run: npm run verify -- ${{ inputs.network }}
```

```suggestion
        run: sleep 10 && npm run verify -- ${{ inputs.network }}
```
Give time to explorers to index contracts.